filament/backend/src/vulkan/VulkanContext.cpp

```diff
@@ -634,5 +634,30 @@ void createDepthBuffer(VulkanContext& context, VulkanSurfaceContext& surfaceCont
     flushWorkCommandBuffer(context);
 }
 
+VkImageLayout getTextureLayout(TextureUsage usage) {
+    // Filament sometimes samples from depth while it is bound to the current render target, (e.g.
+    // SSAO does this while depth writes are disabled) so let's keep it simple and use GENERAL for
+    // all depth textures.
```

Comment:
I would argue that the common case is not that -- i.e. the default depth buffer is never sampled from. Not sure if it's worth handling, but maybe update the comment?

Reply:
Actually, I've read somewhere that on some architectures this could disable depth buffer compression.